File: SecondSem/CN/LAB/Python/ConnectionOrientedFullDuplex_3/Duplexclient.py
import socket


def main():
    try:
        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
            s.connect(("localhost", 9999))
            print("Connected to server...")

            in_from_server = s.makefile("r")
            out_to_server = s.makefile("w")
            in_from_user = input

            while True:
                client_message = input("Client: ")
                out_to_server.write(client_message + "\n")
                out_to_server.flush()

                server_message = in_from_server.readline().strip()
                if server_message == "bye":
                    break
                print("Server: " + server_message)

    except Exception as e:
        print(e)


if __name__ == "__main__":
    main()

File: SecondSem/CN/LAB/Java/ConnectionOrientedFullDuplex_3/DuplexClient.java
import java.io.*;
import java.net.*;

public class DuplexClient {
    public static void main(String[] args) {
        try {
            Socket socket = new Socket("localhost", 9999);
            System.out.println("Connected to server...");

            BufferedReader inFromServer = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            PrintWriter outToServer = new PrintWriter(socket.getOutputStream(), true);
            BufferedReader inFromUser = new BufferedReader(new InputStreamReader(System.in));

            String serverMessage, clientMessage;
            while (true) {
                System.out.print("Client: ");
                clientMessage = inFromUser.readLine();
                outToServer.println(clientMessage);

                serverMessage = inFromServer.readLine();
                if (serverMessage.equals("bye")) {
                    break;
                }
                System.out.println("Server: " + serverMessage);
            }

            socket.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
